engines/group/app/models/role.rb

```diff
@@ -0,0 +1,52 @@
+class Role
+  include Comparable
+
+  attr_reader :index
+
+  class InvalidRole < ArgumentError; end
+
+  VALUES = { admin: 1, member: 2, waiting: 3 }
+
+  def self.from_index(index)
+    case index
+    when 1
+      new(:admin)
+    when 2
+      new(:member)
+    when 3
+      new(:waiting)
+    end
```

Comment:
I found this `case` statement and `VALUES` constant like redundant. Any way to have the `:symbol` > `integer` mapping only in one place?